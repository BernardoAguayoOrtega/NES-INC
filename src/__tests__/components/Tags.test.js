// import react
import React from 'react';
// import shallow
import { mount } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
// import router
import { HashRouter as Router } from 'react-router-dom';
// import context provider
import { ContextProvider } from '../../utils/Context/index.jsx';
// import Tag component
import Tag from '../../components/Tags/index.jsx';

describe('<Tags />', () => {
  const wrapper = mount(
    <ContextProvider>
      <Router>
        <Tag />
      </Router>
    </ContextProvider>,
  );

  it('is the Search rendering?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is there a searchformOff class?', () => {
    expect(wrapper.find('.tag').exists()).toBe(true);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
