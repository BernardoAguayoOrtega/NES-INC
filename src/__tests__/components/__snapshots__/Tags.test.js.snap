// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> matches the snap shot 1`] = `
<ContextProvider>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Tags>
        <Link
          onClick={[Function]}
          to="/"
        >
          <LinkAnchor
            href="#/"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              href="#/"
              onClick={[Function]}
            >
              <div
                aria-label="tag"
                className="tag"
              />
            </a>
          </LinkAnchor>
        </Link>
      </Tags>
    </Router>
  </HashRouter>
</ContextProvider>
`;
